auth/service.go

```diff
@@ -0,0 +1,59 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	authclient "github.com/fabric8-services/fabric8-auth-client/auth"
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/goasupport"
+
+	goaclient "github.com/goadesign/goa/client"
+)
+
+type AuthService interface {
+	RequireScope(ctx context.Context, resourceID, requiredScope string) error
+}
+
+func NewAuthService(authURL string) (AuthService, error) {
+	u, err := url.Parse(authURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &auth{u}, nil
+}
+
+type auth struct {
```

Comment:
The name `auth` is a little generic.. how about we call it `authServiceImplementation` or something like that.

Reply:
Done in https://github.com/fabric8-services/fabric8-common/pull/62/commits/1076d9185c0ec1a4d3d2e65c316aa65346099d9a